tests/rados/test_osd_rebalance.py

```diff
@@ -0,0 +1,163 @@
+            pool_name=pool["pool_name"], pg_num=pool["pg_num"]
+        ):
+            log.error(f"Failed to create the {pool['pool_name']} Pool")
+            return 1
+        rados_obj.change_recover_threads(config=pool, action="set")
+        acting_pg_set = rados_obj.get_pg_acting_set(pool_name=pool["pool_name"])
+        log.info(f"Acting set {acting_pg_set}")
+        if not acting_pg_set:
+            log.error("Failed to retrieve acting pg set")
+            return 1
+        osd_id = acting_pg_set[0]
+        host = rados_obj.fetch_host_node(daemon_type="osd", daemon_id=osd_id)
+        if not host:
+            log.error("Failed to fetch host details")
+            return 1
+
+        # fetch container id
+        out, _ = host.exec_command(sudo=True, cmd="podman ps --format json")
+        container_id = [
+            item["Names"][0]
+            for item in json.loads(out.read().decode())
+            if f"osd.{osd_id}" in item["Command"]
+        ][0]
+        if not container_id:
+            log.error("Failed to retrieve container id")
+            return 1
+        # fetch device path by osd_id
+        vol_out, _ = host.exec_command(
+            sudo=True,
+            cmd=f"podman exec {container_id} ceph-volume lvm list --format json",
+        )
+        volume_out = vol_out.read().decode()
+        dev_path = [
+            v[0]["devices"][0]
+            for k, v in json.loads(volume_out).items()
+            if str(k) == str(osd_id)
+        ][0]
+        if not dev_path:
+            log.error("Failed to get device path")
+            return 1
+        log.info(f"dev_path  : {dev_path}")
+        log.info(f"host.hostname  : {host.hostname}")
+        if not rados_obj.set_osd_devices_unamanged():
+            log.error("Failed to set osd devices unamnaged.")
+            return 1
+        if not rados_obj.bench_write(**pool):
+            log.error("Failed to write objects into the Pool")
+            return 1
+        if not rados_obj.set_osd_out(osd_id):
+            log.error("Failed to set osd out")
+            return 1
+        if not wait_for_clean_pg_sets(rados_obj):
+            return 1
+        if not rados_obj.remove_osd(osd_id):
+            log.error("Failed to remove osd")
+            return 1
+        if not wait_for_clean_pg_sets(rados_obj):
+            return 1
+        if not rados_obj.zap_device(host.hostname, dev_path):
+            log.error("Failed to zap device.")
+            return 1
+        if not wait_for_device(host, container_id, osd_id, action="remove"):
+            log.error("Failed to remove OSD from host.")
+            return 1
+        if not rados_obj.add_osd(host.hostname, dev_path, osd_id):
+            log.error("Failed to add osd.")
+            return 1
+        if not wait_for_device(host, container_id, osd_id, action="add"):
+            log.error("Failed to add OSD to host.")
+            return 1
+        if not wait_for_clean_pg_sets(rados_obj):
```

Comment:
We would have marked the OSD's to be in unmanaged state ( not managed by cephadm ) in the beginning of test, due to code implementation challenges. 

I feel we can set back the OSD daemons to be managed again by cephadm once the OSD has been added back into the cluster.

Reply:
done